crates/solvers/src/boundary/legacy.rs

```diff
@@ -114,14 +114,14 @@ fn to_boundary_auction(
                 (
                     address.0,
                     TokenInfoModel {
-                        decimals: info.decimals,
                         alias: info.symbol.clone(),
                         external_price: info
                             .reference_price
                             .map(|price| price.0 .0.to_f64_lossy() / 1e18),
                         internal_buffer: Some(info.available_balance),
                         accepted_for_internalization: info.trusted,
-                        ..Default::default()
+                        decimals: if address.0 == weth.0 { Some(18) } else { None },
+                        normalize_priority: Some(u64::from(address.0 == weth.0)),
```

Comment:
Quasimodo's `/solve` request will crash if it doesn't know the decimals of the reference token. I crudely made the native token the reference token and hardcoded its decimals to 18 until we properly fetch all the token meta data.